### PR TITLE
bridge2: use "completions" api for assistant

### PR DESCRIPTION
**Pull request by mgood**
Switch from "chat completion" to "completions"
API to have more direct control over the format.

This includes a basic template file as well to
customize the prompt format.

The template is exposed for live editing to test
changes while the service is running.

See #190

### CODE REVIEW

File: images/bridge2/assistant/assistant.go [hunk 1]
```diff
@@ -109,24 +110,22 @@ Overall {} is a powerful system that can help humans with a wide range of tasks
...
-	req := ChatCompletionRequest{
+	prompt, err := prompts.Render("complete", map[string]any{
+		"SystemMessage": systemMessage,
+		"UserInput":     input,
+	})
+	if err != nil {
+		return "", err
+	}
+	req := CompletionRequest{
+		// Model:
```
Comment (mgood) [inline images/bridge2/assistant/assistant.go:124]: Is there a value we want here for the model?

**Review by mgood — COMMENTED** [review]
This seems to produce at least some response, though it does include the name "bridge" at the beginning.

Screenshot 2024-03-22 at 4 40 48 PM

File: images/bridge2/assistant/assistant.go (hunk 1 above)
Comment (ajbouh) [reply images/bridge2/assistant/assistant.go:124]: no, nothing specific atm

**Comment by ajbouh** [comment]
> This seems to produce at least some response, though it does include the name "bridge" at the beginning.
> 
> Screenshot 2024-03-22 at 4 40 48 PM

Can fix this by including bridge's name in the prompt we send over

File: images/bridge2/assistant/prompts/complete.tmpl [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{{.SystemMessage}}
+
+USER: {{.UserInput}}
```
Comment (ajbouh) [inline images/bridge2/assistant/prompts/complete.tmpl:3]:
```suggestion
USER: {{.UserInput}}
PROMPT:
```
(but *no* trailing newline)

**Comment by mgood** [comment]
I'm going to start on changes to add/remove assistants, but this seems to work reasonably after that latest change to add the name in the prompt.

**Comment by ajbouh** [comment]
great, shall we merge it and do the rest in the next PR?

**Comment by ajbouh** [comment]
also, do you want to remove the chat completion api or leave it for now?

**Comment by mgood** [comment]
> also, do you want to remove the chat completion api or leave it for now?

yeah, I can remove it, it's easy enough to add-back if we need it again

**Comment by mgood** [comment]
> great, shall we merge it and do the rest in the next PR?

yeah, sounds good

**Pull request merged**